invenio_rdm_records/fixtures/vocabularies.py

```diff
@@ -106,84 +109,249 @@ def __iter__(self):
 
 
 #
-# Fixture
+# Exceptions
 #
-class VocabulariesFixture:
-    """Vocabularies fixture."""
+class ConflictingFixturesError(Exception):
+    """Exception when multiple modules provide same subject."""
 
-    def __init__(self, identity, search_paths, filename):
-        """Initialize the fixture."""
-        self._search_paths = search_paths
-        self._filename = filename
+    def __init__(self, errors):
+        """Constructor."""
+        message = "\n".join(errors)
+        super().__init__(message)
+        self.errors = errors
+
+
+#
+# Fixture
+#
+class VocabulariesFixtures:
+    """Concept of Vocabulary fixtures across locations.
+
+    This class' responsibility is to load different vocabularies in
+    this priority order: app_data, then extensions and then this package.
+
+    Earlier vocabularies in this hierarchy are chosen first. But in the case
+    of subvocabularies (e.g. subject types), having loaded a subvocabulary
+    before, shouldn't prevent loading another one. Only if the same
+    subvocabulary is encountered again should it be ignored.
+
+    Concretely, having loaded MeSH subjects shouldn't prevent loading FAST
+    subjects even though they are both under the "subjects" vocabulary.
+    Another MeSH subject encountered down the hierarchy is ignored however.
+    """
+
+    def __init__(self, identity, app_data_folder=None, pkg_data_folder=None,
+                 filename="vocabularies.yaml", delay=True):
+        """Constructor.
+
+        identity: Identity to use when loading
+        app_data_folder: Path object to instance data folder
+        pkg_data_folder: Path object to this package's data folder.
+                         Defaults to `./data` and really only changeable for
+                         tests.
+        filename: vocabularies filename to check at each location
+        """
         self._identity = identity
+        # Path("./app_data") assumes app_data is in current working directory
```

Comment:
remove?
```suggestion
```

Reply:
It applies to the line below it, so would keep. Basically, `Path('app_data/')` relies on wider knowledge of the structure of our application implicitly: it knows that when `invenio-cli rdm-records fixtures` is called, it is called within the main directory of the instance, where there is an `app_data/` folder. For something so tucked away in the code base it's good to flag it.